artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/impl/BridgeImpl.java

```diff
@@ -634,16 +641,26 @@ public HandleStatus handle(final MessageReference ref) throws Exception {
             }
 
             if (server.hasBrokerBridgePlugins()) {
-               server.callBrokerBridgePlugins(plugin -> plugin.afterDeliverBridge(this, ref, status));
+               final HandleStatus finalStatus = status;
+               server.callBrokerBridgePlugins(plugin -> plugin.afterDeliverBridge(this, ref, finalStatus));
+               return finalStatus;
             }
 
             return status;
          } catch (Exception e) {
             // If an exception happened, we must count down immediately
             pendingAcks.countDown();
-            throw e;
+            exception = e;
+         }
+
+      } finally {
+         bridgeLock.unlock();
+         if (exception != null) {
```

Comment:
This is wholly unneeded as a finally block will run regardless of an exception so the local is assured to always be unlocked.